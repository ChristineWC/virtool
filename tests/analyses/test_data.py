from virtool_core.models.enums import QuickAnalyzeWorkflow

from virtool.data.utils import get_data_from_app
from tests.users.test_data import users_data


async def test_create(app, mongo, snapshot, static_time, users_data):
    """
    Tests that an analysis is created with the expected fields.
    """
    subtractions = ["subtraction_1", "subtraction_2"]

    await users_data.create(password="hello_world", handle="bill")
    user_info = await mongo.users.find_one({"handle": "bill"})

    async with mongo.create_session() as session:
        await mongo.indexes.insert_one(
            {
                "_id": "test_index",
                "version": 11,
                "ready": True,
                "reference": {"id": "test_ref"},
            }
        )

        await mongo.references.insert_one(
            {
                "_id": "test_ref",
                "name": "ref_name",
                "data_type": "genome",
            }
        )

        await mongo.subtraction.insert_many(
            [
                {
                    "_id": "subtraction_1",
                    "name": "s1",
                },
                {
                    "_id": "subtraction_2",
                    "name": "s2",
                },
            ],
            session=session,
        )

    await get_data_from_app(app).analyses.create(
        "test_sample",
        "test_ref",
        subtractions,
        user_info["_id"],
        QuickAnalyzeWorkflow.nuvs,
        "test_job",
        0,
        analysis_id="test_analysis",
    )

    assert await mongo.analyses.find_one({"_id": "test_analysis"}) == snapshot
