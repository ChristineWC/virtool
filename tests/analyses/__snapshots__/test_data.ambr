# name: test_create[mongo]
  <class 'dict'> {
    '_id': 'test_analysis',
    'files': <class 'list'> [
    ],
    'index': <class 'dict'> {
      'id': 'test_index',
      'version': 11,
    },
    'ready': False,
    'reference': <class 'dict'> {
      'data_type': 'genome',
      'id': 'test_ref',
      'name': 'Test Reference',
    },
    'results': None,
    'sample': <class 'dict'> {
      'id': 'test_sample',
    },
    'space': <class 'dict'> {
      'id': 0,
    },
    'subtractions': <class 'list'> [
      <class 'dict'> {
        '_id': 'subtraction_1',
        'name': 'Subtraction 1',
      },
      <class 'dict'> {
        '_id': 'subtraction_2',
        'name': 'Subtraction 2',
      },
    ],
    'user': <class 'dict'> {
      'administrator': False,
      'handle': 'leeashley',
      'id': 'bf1b993c',
    },
    'workflow': 'nuvs',
  }
---
# name: test_create[obj]
  <class 'dict'> {
    'files': <class 'list'> [
    ],
    'id': 'test_analysis',
    'index': <class 'dict'> {
      'id': 'test_index',
      'version': 11,
    },
    'job': <class 'dict'> {
      'archived': False,
      'id': 'fb085f7f',
      'progress': 0,
      'stage': None,
      'state': <JobState.WAITING: 'waiting'>,
      'user': <class 'dict'> {
        'administrator': False,
        'handle': 'leeashley',
        'id': 'bf1b993c',
      },
      'workflow': 'nuvs',
    },
    'ready': False,
    'reference': <class 'dict'> {
      'data_type': <ReferenceDataType.genome: 'genome'>,
      'id': 'test_ref',
      'name': 'Test Reference',
    },
    'results': None,
    'sample': <class 'dict'> {
      'id': 'test_sample',
    },
    'subtractions': <class 'list'> [
      <class 'dict'> {
        'id': 'subtraction_1',
        'name': 'Subtraction 1',
      },
      <class 'dict'> {
        'id': 'subtraction_2',
        'name': 'Subtraction 2',
      },
    ],
    'user': <class 'dict'> {
      'administrator': False,
      'handle': 'leeashley',
      'id': 'bf1b993c',
    },
    'workflow': 'nuvs',
  }
---
# name: test_create_analysis_id[mongo]
  <class 'dict'> {
    '_id': 'fb085f7f',
    'files': <class 'list'> [
    ],
    'index': <class 'dict'> {
      'id': 'test_index',
      'version': 11,
    },
    'ready': False,
    'reference': <class 'dict'> {
      'data_type': 'genome',
      'id': 'test_ref',
      'name': 'Test Reference',
    },
    'results': None,
    'sample': <class 'dict'> {
      'id': 'test_sample',
    },
    'space': <class 'dict'> {
      'id': 0,
    },
    'subtractions': <class 'list'> [
      <class 'dict'> {
        '_id': 'subtraction_1',
        'name': 'Subtraction 1',
      },
      <class 'dict'> {
        '_id': 'subtraction_2',
        'name': 'Subtraction 2',
      },
    ],
    'user': <class 'dict'> {
      'administrator': False,
      'handle': 'leeashley',
      'id': 'bf1b993c',
    },
    'workflow': 'nuvs',
  }
---
# name: test_create_analysis_id[obj]
  <class 'dict'> {
    'files': <class 'list'> [
    ],
    'id': 'fb085f7f',
    'index': <class 'dict'> {
      'id': 'test_index',
      'version': 11,
    },
    'job': <class 'dict'> {
      'archived': False,
      'id': '7cf872dc',
      'progress': 0,
      'stage': None,
      'state': <JobState.WAITING: 'waiting'>,
      'user': <class 'dict'> {
        'administrator': False,
        'handle': 'leeashley',
        'id': 'bf1b993c',
      },
      'workflow': 'nuvs',
    },
    'ready': False,
    'reference': <class 'dict'> {
      'data_type': <ReferenceDataType.genome: 'genome'>,
      'id': 'test_ref',
      'name': 'Test Reference',
    },
    'results': None,
    'sample': <class 'dict'> {
      'id': 'test_sample',
    },
    'subtractions': <class 'list'> [
      <class 'dict'> {
        'id': 'subtraction_1',
        'name': 'Subtraction 1',
      },
      <class 'dict'> {
        'id': 'subtraction_2',
        'name': 'Subtraction 2',
      },
    ],
    'user': <class 'dict'> {
      'administrator': False,
      'handle': 'leeashley',
      'id': 'bf1b993c',
    },
    'workflow': 'nuvs',
  }
---
